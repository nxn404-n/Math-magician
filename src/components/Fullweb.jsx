import { Outlet, Link } from 'react-router-dom';
import '../styles/Fullweb.css';

export default function Fullweb() {
  return (
    <>
      <header className="headerWebsite">
        <h2>Math Magicians</h2>
        <nav>
          <ul>
            <li><Link to="/">Home</Link></li>
            <li><Link to="/calculator">Calculator</Link></li>
            <li><Link to="/quote">Quote</Link></li>
          </ul>
        </nav>
      </header>
      <Outlet />
    </>
  );
}
