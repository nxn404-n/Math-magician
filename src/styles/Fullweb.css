.headerWebsite {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem;
  box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
  background: #8a8585;
}

.headerWebsite h2 {
  font-size: 2rem;
  font-weight: 700;
  color: aliceblue;
}

nav ul {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  list-style: none;
}

nav li {
  margin-left: 1rem;
}

nav li a {
  font-size: 1.2rem;
  font-weight: 500;
  color: aliceblue;
  text-decoration: none;
  padding: 0.5rem;
  border-radius: 0.25rem;
  transition: all 0.2s ease;
}

nav li a:hover {
  background-color: #333;
  color: #fff;
}
