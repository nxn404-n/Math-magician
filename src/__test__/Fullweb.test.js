import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Fullweb from '../components/Fullweb';

describe('Fullweb component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Fullweb />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
