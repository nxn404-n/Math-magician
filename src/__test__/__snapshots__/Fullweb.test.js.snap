// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fullweb component renders correctly 1`] = `
<header
  className="headerWebsite"
>
  <h2>
    Math Magicians
  </h2>
  <nav>
    <ul>
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
